clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3143,60 +3143,6 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return !SemaRef.getASTContext().hasSameType(FD->getType(), Ty);
   }
 
-  // Sets a flag if the kernel is a parallel_for that calls the
-  // free function API "this_item".
-  void setThisItemIsCalled(FunctionDecl *KernelFunc) {
-    if (getKernelInvocationKind(KernelFunc) != InvokeParallelFor)
-      return;
-
-    // The call graph for this translation unit.
-    CallGraph SYCLCG;
-    SYCLCG.addToCallGraph(SemaRef.getASTContext().getTranslationUnitDecl());
-    using ChildParentPair =
-        std::pair<const FunctionDecl *, const FunctionDecl *>;
-    llvm::SmallPtrSet<const FunctionDecl *, 16> Visited;
-    llvm::SmallVector<ChildParentPair, 16> WorkList;
-    WorkList.push_back({KernelFunc, nullptr});
-
-    while (!WorkList.empty()) {
-      const FunctionDecl *FD = WorkList.back().first;
-      WorkList.pop_back();
-      if (!Visited.insert(FD).second)
-        continue; // We've already seen this Decl
-
-      // Check whether this call is to free functions (sycl::this_item(),
-      // this_id, etc.).
-      if (Util::isSyclFunction(FD, "this_id")) {
```

Comment:
I think you can also delete `isSyclFunction `declaration and definition in this file. I believe it is only used for this purpose